src/Uno.UI.Composition/Composition/SkiaCompositionSurface.skia.cs

```diff
@@ -1,27 +1,43 @@
 ﻿#nullable enable
 
-
 using SkiaSharp;
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Runtime.CompilerServices;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 using Uno.Extensions;
 using Uno.Foundation.Logging;
+using Uno.UI.Dispatching;
 using Windows.Graphics;
 
 namespace Microsoft.UI.Composition
 {
-	internal partial class SkiaCompositionSurface : ICompositionSurface
+	internal partial class SkiaCompositionSurface : CompositionObject, ICompositionSurface
 	{
-		private SKImage? _image;
+		// Don't set this field directly. Use SetFrameProviderAndOnFrameChanged instead.
+		private ImageFrameProvider? _frameProvider;
+
+		// Unused: But intentionally kept!
+		// This is here to keep the Action lifetime the same as SkiaCompositionSurface.
+		// i.e, only cause the Action to be GC'ed if SkiaCompositionSurface is GC'ed.
+		private Action? _onFrameChanged;
```

Comment:
A possibly better solution is to make `ImageFrameProvider` take a `WeakReference<SkiaCompositionSurface>` instead of a `WeakReference<Action>`?

Reply:
Maybe will keep it as is for now, unless there is a strong opinion to the other approach.